From 0fa9d07286fa22499c4502fa374a6f93c2b12742 Mon Sep 17 00:00:00 2001
From: Brendan King <Brendan.King@imgtec.com>
Date: Fri, 24 Nov 2017 16:26:10 +0000
Subject: [PATCH 03/12] glamor: if no DRI2, don't call DRI2CloseScreen

The ms_dri2_screen_init function checks for DRI2 before performing
DRI2 initialisation, but ms_dri2_close_screen wasn't checking before
calling DRI2CloseScreen.
---
 hw/xfree86/drivers/modesetting/dri2.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hw/xfree86/drivers/modesetting/dri2.c b/hw/xfree86/drivers/modesetting/dri2.c
index 8d1b742ef..9b202862b 100644
--- a/hw/xfree86/drivers/modesetting/dri2.c
+++ b/hw/xfree86/drivers/modesetting/dri2.c
@@ -1109,7 +1109,8 @@ ms_dri2_screen_init(ScreenPtr screen)
 void
 ms_dri2_close_screen(ScreenPtr screen)
 {
-    DRI2CloseScreen(screen);
+    if (xf86LoaderCheckSymbol("DRI2Version"))
+        DRI2CloseScreen(screen);
 }
 
 #endif /* GLAMOR_HAS_GBM */
-- 
2.25.1

